src/gan/discriminator/model.py

```diff
@@ -1,72 +1,213 @@
 import torch
 import torch.nn as nn
-import torch.nn.functional as F
-
-
-# module for childsumtreelstm
-class ChildSumTreeLSTM(nn.Module):
-    def __init__(self, in_dim, mem_dim):
-        super(ChildSumTreeLSTM, self).__init__()
-        self.in_dim = in_dim
-        self.mem_dim = mem_dim
-        self.ioux = nn.Linear(self.in_dim, 3 * self.mem_dim)
-        self.iouh = nn.Linear(self.mem_dim, 3 * self.mem_dim)
-        self.fx = nn.Linear(self.in_dim, self.mem_dim)
-        self.fh = nn.Linear(self.mem_dim, self.mem_dim)
-
-    def node_forward(self, inputs, child_c, child_h):
-        child_h_sum = torch.sum(child_h, dim=0, keepdim=True)
-
-        iou = self.ioux(inputs) + self.iouh(child_h_sum)
-        i, o, u = torch.split(iou, iou.size(1) // 3, dim=1)
-        i, o, u = torch.sigmoid(i), torch.sigmoid(o), torch.tanh(u)
-
-        f = torch.sigmoid(
-            self.fh(child_h) +
-            self.fx(inputs).repeat(len(child_h), 1)
-        )
-        fc = torch.mul(f, child_c)
+from torch.utils.data import DataLoader
+from tqdm import tqdm
 
-        c = torch.mul(i, u) + torch.sum(fc, dim=0, keepdim=True)
-        h = torch.mul(o, torch.tanh(c))
-        return c, h
 
-    def forward(self, tree, inputs):
-        for idx in range(tree.num_children):
-            self.forward(tree.children[idx], inputs)
+class QueryGAN_Discriminator_CNN(nn.Module):
+    def __init__(self, args, vocab, vectors, output_shape):
+        super(QueryGAN_Discriminator_CNN, self).__init__()
 
-        if tree.num_children == 0:
-            child_c = inputs[0].detach().new(1, self.mem_dim).fill_(0.).requires_grad_()
-            child_h = inputs[0].detach().new(1, self.mem_dim).fill_(0.).requires_grad_()
-        else:
-            child_c, child_h = zip(* map(lambda x: x.state, tree.children))
-            child_c, child_h = torch.cat(child_c, dim=0), torch.cat(child_h, dim=0)
+        self.args = args
```

Comment:
If the `args` dictionary contains required arguments (like the `filter_sizes`), maybe pass these separately to the constructor?